packages/contracts/src/plugins/governance/majority-voting/token/TokenVotingSetup.sol

```diff
@@ -275,4 +232,25 @@ contract TokenVotingSetup is PluginSetup {
         );
         return success && data.length == 0x20;
     }
+
+    /// @notice Unsatisfiably determines if the token is an IVotes interface.
+    /// @dev Many tokens don't use ERC165 even though they still support IVotes.
+    function supportsIVotesInterface(address token) private view returns (bool) {
+        (bool success1, bytes memory data1) = token.staticcall(
+            abi.encodeWithSelector(IVotesUpgradeable.getPastTotalSupply.selector, 0)
+        );
+        (bool success2, bytes memory data2) = token.staticcall(
+            abi.encodeWithSelector(IVotesUpgradeable.getVotes.selector, address(this))
+        );
+        (bool success3, bytes memory data3) = token.staticcall(
+            abi.encodeWithSelector(IVotesUpgradeable.getPastVotes.selector, address(this), 0)
+        );
+
+        return (success1 &&
+            data1.length == 0x20 &&
+            success2 &&
+            data2.length == 0x20 &&
+            success3 &&
+            data3.length == 0x20);
+    }
```

Comment:
Slimmer variant with early returns.

```suggestion
        (bool success, bytes memory data) = token.staticcall(
            abi.encodeWithSelector(IVotesUpgradeable.getPastTotalSupply.selector, 0)
        );
        if(!success || data.length != 0x20) return false;

        (success, data) = token.staticcall(
            abi.encodeWithSelector(IVotesUpgradeable.getVotes.selector, address(this))
        );
        if(!success || data.length != 0x20) return false;

        (success, data) = token.staticcall(
            abi.encodeWithSelector(IVotesUpgradeable.getPastVotes.selector, address(this), 0)
        );

        return success && data.length == 0x20;
    }
```

Reply:
I vote for readability and not gas cost effectiveness in this case.  I will resolve this as the current code looks more readeable.